tests/database_test.cpp

```diff
@@ -208,3 +208,216 @@ TEST_F(DatabaseTest, NoConfigTransaction) {
          ASSERT_EQ(0, tx.Read("bob").second);
        }});
 }
+
+// [Secondary Index TDD] ------------------------------------------------------
+TEST_F(DatabaseTest, CreateTable) {
+  db_ = std::make_unique<LineairDB::Database>();
+  bool success = db_->CreateTable("users");
+  ASSERT_TRUE(success);
+  bool duplicated = db_->CreateTable("users");
+  ASSERT_FALSE(duplicated);
+}
+
+//CreateSecondaryIndex("table_name", "index_name", "UNIQUE", "NOT_NULL")
+TEST_F(DatabaseTest, CreateSecondaryIndex) {
+  db_ = std::make_unique<LineairDB::Database>();
+  bool success = db_->CreateTable("users");
+  bool index_success = db_->CreateSecondaryIndex("users", "name", "UNIQUE", "NOT_NULL");
+  ASSERT_TRUE(index_success);
+}
+
+//without constraint
+TEST_F(DatabaseTest, CreateSecondaryIndexWithoutConstraints) {
+  db_ = std::make_unique<LineairDB::Database>();
+  ASSERT_TRUE(db_->CreateTable("users"));
+  ASSERT_TRUE(db_->CreateSecondaryIndex("users", "nickname"));
+}
+
+// UNIQUE only
+TEST_F(DatabaseTest, CreateSecondaryIndexUniqueOnly) {
+  db_ = std::make_unique<LineairDB::Database>();
+  ASSERT_TRUE(db_->CreateTable("users"));
+  ASSERT_TRUE(db_->CreateSecondaryIndex("users", "email", "UNIQUE"));
+  
+}
+
+// NOT_NULL only
+TEST_F(DatabaseTest, CreateSecondaryIndexNotNullOnly) {
+  db_ = std::make_unique<LineairDB::Database>();
+  ASSERT_TRUE(db_->CreateTable("users"));
+  ASSERT_TRUE(db_->CreateSecondaryIndex("users", "phone", "NOT_NULL"));
+}
+
+// constraint order
+TEST_F(DatabaseTest, CreateSecondaryIndexConstraintOrderIndifference) {
+  db_ = std::make_unique<LineairDB::Database>();
+  ASSERT_TRUE(db_->CreateTable("users"));
+  ASSERT_TRUE(db_->CreateSecondaryIndex("users", "address", "NOT_NULL", "UNIQUE"));
```

Comment:
You wrote this test case (checking that changing the arguments order has no effect), which suggests that the design is not good (i.e., it is a non-intuitive API).